velox/functions/prestosql/SIMDJsonFunctions.h

```diff
@@ -71,52 +71,39 @@ struct SIMDJsonArrayContainsFunction {
 
     result = false;
     for (auto&& v : jsonDoc) {
-      try {
-        if constexpr (std::is_same_v<TInput, bool>) {
-          if (v.type() == simdjson::ondemand::json_type::boolean &&
-              v.get_bool() == value) {
-            result = true;
-            break;
-          }
-        } else if constexpr (std::is_same_v<TInput, int64_t>) {
-          if (v.type() == simdjson::ondemand::json_type::number &&
-              v.get_number_type() ==
-                  simdjson::ondemand::number_type::signed_integer &&
-              v.get_int64() == value) {
-            result = true;
-            break;
-          }
-        } else if constexpr (std::is_same_v<TInput, double>) {
-          if (v.type() == simdjson::ondemand::json_type::number &&
-              v.get_number_type() ==
-                  simdjson::ondemand::number_type::floating_point_number &&
-              v.get_double() == value) {
+      if constexpr (std::is_same_v<T, bool>) {
+        bool boolValue;
+        if (v.type() == simdjson::ondemand::json_type::boolean &&
+            !v.get_bool().get(boolValue) && boolValue == value) {
```

Comment:
To match the old code, if the error here is not `INCORRECT_TYPE` or `NUMBER_ERROR`, we should return null immediately.  We maybe need to check this behavior to match what Java is doing.

Reply:
My reading of the Java code suggests that there is no 'early termination' logic. Java returns NULL only on exception, end of document, end of array.

presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```
        try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
            if (parser.nextToken() != START_ARRAY) {
                return null;
            }

            while (true) {
                JsonToken token = parser.nextToken();
                if (token == null) {
                    return null;
                }
                if (token == END_ARRAY) {
                    return false;
                }
                parser.skipChildren();

                if ((token == VALUE_NUMBER_INT) &&
                        ((parser.getNumberType() == NumberType.INT) || (parser.getNumberType() == NumberType.LONG)) &&
                        (parser.getLongValue() == value)) {
                    return true;
                }
            }
        }
        catch (IOException e) {
            return null;
        }
```

Reply:
There is a sanity check for 'double' though. This is missing in Velox. I'll make a separate PR to add.

```
        if (!Doubles.isFinite(value)) {
            return false;
        }
```